src/engine/loopingcontrol.cpp

```diff
@@ -385,12 +385,28 @@ double LoopingControl::nextTrigger(bool reverse,
                 // should be moved with it
                 *pTarget = seekInsideAdjustedLoop(currentSample,
                         m_oldLoopSamples.start, loopSamples.start, loopSamples.end);
+            } else {
+                bool movedOut = false;
+                // Ceck if we have moved out of the loop, before we could enable it
+                if (reverse) {
+                    if (loopSamples.start > currentSample) {
+                        movedOut = true;
+                    }
+                } else {
+                    if (loopSamples.end < currentSample) {
+                        movedOut = true;
+                    }
+                }
+                if (movedOut) {
+                    *pTarget = seekInsideAdjustedLoop(currentSample,
+                            loopSamples.start, loopSamples.start, loopSamples.end);
+                }
             }
             m_oldLoopSamples = loopSamples;
             if (*pTarget != kNoTrigger) {
                 // jump immediately
-                qDebug() << currentSample <<
-                        m_oldLoopSamples.start << loopSamples.start << loopSamples.end;
+                //qDebug() << currentSample <<
```

Comment:
This commented out debug output is confusing, because m_oldLoopSamples == loopSamples after the assignment and why should we print the pTarget pointer? I think it is better to simply remove this commented out piece of code.

There is also a minor typo in another comment (line 390) above if you touch this file: "Ceck" -> "Check"